src/main/scala/nl/knaw/dans/easy/stage/EasyStageDataset.scala

```diff
@@ -108,6 +129,8 @@ object EasyStageDataset {
       Try { dir.listFiles().toList }.flatMap(_.map(visit).collectResults.map(_ => ()))
     }
 
+    def getBagRelativePath(path: Path) = s.bagitDir.toPath.relativize(path)
```

Comment:
that's why! It's a big nested function, so you need to be precise with the types 😉 
maybe we can flatten (un-nest) the function by making the inner-functions private in the outer scope? might be beyond the scope of this issue.